core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -174,9 +175,13 @@
    private volatile boolean stopping = false;
    private boolean transactional;
    private boolean batchingEnabled;
+   private final ContextBuilder contextBuilder;
+   private final ContextBuilder pferContextBuilder;
 
    public CacheImpl(String name) {
       this.name = name;
+      this.contextBuilder = this::getInvocationContextWithImplicitTransaction;
```

Comment:
I would prefer to initialize these inline in field declaration rather than constructor... If I am not nitpicking too much...